lib/ansible/modules/cloud/amazon/inspector_target.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+# Copyright (c) 2018 Dennis Conrad for Sainsbury's
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: inspector_target
+short_description: Create, Update and Delete Amazon Inspector Assessment
+                   Targets
+description: Creates, updates, or deletes Amazon Inspector Assessment Targets
+             and manages the required Resource Groups.
+version_added: "2.6"
+author: "Dennis Conrad (@dennisconrad)"
+options:
+  name:
+    description:
+      - The user-defined name that identifies the assessment target.  The name
+        must be unique within the AWS account.
+    required: true
+  state:
+    description:
+      - The state of the assessment target.
+    choices:
+      - absent
+      - present
+    default: present
+  tags:
+    description:
+      - Tags of the EC2 instances to be added to the assessment target.
+      - Required if C(state=present).
+extends_documentation_fragment:
+  - aws
+  - ec2
+requirements:
+  - boto3
+  - botocore
+'''
+
+EXAMPLES = '''
+- name: Create my_target Assessment Target
+  inspector_target:
+    name: my_target
+    tags:
+      role: scan_target
+
+- name: Update Existing my_target Assessment Target with Additional Tags
+  inspector_target:
+    name: my_target
+    tags:
+      env: dev
+      role: scan_target
+
+- name: Delete my_target Assessment Target
+  inspector_target:
+    name: my_target
+    state: absent
+'''
+
+RETURN = '''
+arn:
+  description: The ARN that specifies the Amazon Inspector assessment target.
+  returned: success
+  type: string
+  sample: "arn:aws:inspector:eu-west-1:123456789012:target/0-O4LnL7n1"
+created_at:
+  description: The time at which the assessment target was created.
+  returned: success
+  type: string
+  sample: "2018-01-29T13:48:51.958000+00:00"
+name:
+  description: The name of the Amazon Inspector assessment target.
+  returned: success
+  type: string
+  sample: "my_target"
+resource_group_arn:
+  description: The ARN that specifies the resource group that is associated
+               with the assessment target.
+  returned: success
+  type: string
+  sample: "arn:aws:inspector:eu-west-1:123456789012:resourcegroup/0-qY4gDel8"
+tags:
+  description: The tags of the resource group that is associated with the
+               assessment target.
+  returned: success
+  type: list
+  sample: [{"key": "role", "value": "scan_target"},
```

Comment:
The tags should be returned flattened like other AWS modules:
{"role": "scan_target", "env": "dev"}

You can import boto3_tag_list_to_ansible_dict from ansible.module_utils.ec2 to make that transformation.